### PR TITLE
Refactored some cursed height limit view names

### PR DESCRIPTION
**Pull request by liach**
They were either confusing or plainly wrong

Fixed a few javadoc problems

enbrain: I wrote these docs just to prove how my names make way more sense than yours. ~~Yours were comparable to mojank~~

### CODE REVIEW

**Comment by liach** [comment]
Fyi I made this not because I want to. It's because these misleading names cause great problems when I try to map nether portal stuff.

File: mappings/net/minecraft/world/HeightLimitView.mapping [hunk 1]
```diff
@@ -1,17 +1,82 @@
...
+		COMMENT
+		COMMENT @return a zero-based index
+		COMMENT @see #sectionIndexToCoord(int) the inverse operation <code>sectionIndexToCoord</code>
+		ARG 1 coord
+			COMMENT the section coordinate
+	METHOD method_31604 sectionIndexToCoord (I)I
+		COMMENT Converts a zero-based section index to a section coordinate.
+		COMMENT
+		COMMENT @return a section coordinate
+		COMMENT @see #sectionCoordToIndex(int) the inverse operation the inverse operation <code>sectionCoordToIndex</code>
```
Comment (liach) [inline mappings/net/minecraft/world/HeightLimitView.mapping]:
```suggestion
		COMMENT @see #sectionCoordToIndex(int) the inverse operation <code>sectionCoordToIndex</code>
```
somehow

File: mappings/net/minecraft/world/HeightLimitView.mapping [hunk 2]
```diff
@@ -1,17 +1,82 @@
 CLASS net/minecraft/class_5539 net/minecraft/world/HeightLimitView
-	METHOD method_31597 getTopSectionLimit ()I
-	METHOD method_31600 getTopHeightLimit ()I
+	COMMENT A view with a high limit specification.
```
Comment (enbrain) [inline mappings/net/minecraft/world/HeightLimitView.mapping]:
```suggestion
	COMMENT A view with a height limit specification.
```

File: mappings/net/minecraft/world/HeightLimitView.mapping [hunk 3]
```diff
@@ -1,17 +1,82 @@
...
+		ARG 1 coord
+			COMMENT the section coordinate
+	METHOD method_31604 sectionIndexToCoord (I)I
+		COMMENT Converts a zero-based section index to a section coordinate.
+		COMMENT
+		COMMENT @return a section coordinate
+		COMMENT @see #sectionCoordToIndex(int) the inverse operation <code>sectionCoordToIndex</code>
+		ARG 1 index
+			COMMENT the zero-based section index
+	METHOD method_31605 getDeltaY ()I
```
Comment (enbrain) [inline mappings/net/minecraft/world/HeightLimitView.mapping]: Maybe `getVerticalSize` or `getHeight`?

IMO `world.getDeltaY()` is unclear what `delta` means.

Comment (liach) [reply mappings/net/minecraft/world/HeightLimitView.mapping]: getHeight sounds good. I need to document that mojang calls this `logical_height` in the dimension codec

Comment (liach) [reply mappings/net/minecraft/world/HeightLimitView.mapping]: Turns out logical height is an unrelated concept only used by teleporting/portal logic. I renamed an associated method in world though (called `getHeightLimit` but calls `dimension.getLogicalHeight`)

**Review by enbrain — APPROVED** [review]

File: mappings/net/minecraft/world/HeightLimitView.mapping [hunk 4]
```diff
@@ -1,17 +1,82 @@
...
+	METHOD method_31600 getTopY ()I
+		COMMENT Returns the top Y level, or height, exclusive, of this view.
+		COMMENT
+		COMMENT @implNote This implementation sums up the bottom Y and the height.
+		COMMENT @see #getBottomY()
+		COMMENT @see #getHeight()
 	METHOD method_31601 isOutOfHeightLimit (I)Z
+		COMMENT Checks if {@code y} is out of the height limit of this view.
+		COMMENT
+		COMMENT <P>{@code y} is out of bounds if it's lower than the {@linkplain #getBottomY
```
Comment (YanisBft) [inline mappings/net/minecraft/world/HeightLimitView.mapping]:
```suggestion
		COMMENT <p>{@code y} is out of bounds if it's lower than the {@linkplain #getBottomY
```

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

File: mappings/net/minecraft/world/HeightLimitView.mapping (hunk 4 above)
Comment (liach) [reply mappings/net/minecraft/world/HeightLimitView.mapping]: How did I even make that typo

**Review by jaskarth — APPROVED** [review]

**Pull request merged**